core/src/main/scala/org/apache/spark/resource/ResourceProfile.scala

```diff
@@ -194,7 +194,8 @@ class ResourceProfile(
       throw new SparkException("No executor resource configs were not specified for the " +
         s"following task configs: ${taskResourcesToCheck.keys.mkString(",")}")
     }
-    logInfo(s"Limiting resource is $limitingResource at $taskLimit tasks per executor")
+    val limiting = if (taskLimit == -1) "None" else limitingResource
+    logInfo(s"Limiting resource is $limiting at $taskLimit tasks per executor")
```

Comment:
If you don't have custom executor resources then the limiting resource should always be CPU right? Also the `-1 tasks per executor` really looks like some result from compute overflow, could we skip it if the `taskLimit` is <= 0?

Reply:
yes good idea, if taskLimiit == -1 then just set it to CPU because there weren't any other custom resources.